### PR TITLE
fix: nvidia egress with cached components

### PR DESCRIPTION
**Pull request by alexeldeib**

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 1]
```diff
@@ -303,8 +302,8 @@ installNvidiaDocker() {
     if [ -d "$dst/pkg" ]; then
         if [ -f "$dst/pkg/DEBIAN/control" ]; then
             installed="$(cat "$dst/pkg/DEBIAN/control" | grep Version | cut -d' ' -f 2)"
-            if [ "$version" == "$installed" ]; then
-                echo "skip nvidia-docker2 install, current version $version matches target $target"
+            if [ "$target" == "$installed" ]; then
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:305]: $version is empty here, so this always failed, and we pulled from nvidia at runtime.

**Comment by alexeldeib** [comment]
@anujmaheshwari1 this would be a good case to catch with gpu e2e

**Review by UtheMan — APPROVED** [review]
LGTM

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 2]
```diff
@@ -303,8 +302,8 @@ installNvidiaDocker() {
...
+                echo "skip nvidia-docker2 install, current version $target matches target $target"
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:306]: target matches target, oops

File: e2e/nodebootstrapping_template.json [hunk 3]
```diff
@@ -132,7 +132,7 @@
     "Disable1804SystemdResolved": false,
     "EnableGPUDevicePluginIfNeeded": false,
     "EnableKubeletConfigFile": false,
-    "EnableNvidia": false,
+    "EnableNvidia": true,
```
Comment (anujmaheshwari1) [inline e2e/nodebootstrapping_template.json:135]: set it to false post e2e testing?

**Pull request merged**